x/concentrated-liquidity/incentives.go

```diff
@@ -969,9 +966,11 @@ func (k Keeper) collectIncentives(ctx sdk.Context, sender sdk.AccAddress, positi
 		return sdk.Coins{}, sdk.Coins{}, err
 	}
 
-	err = k.ensurePositionOwner(ctx, sender, position.PoolId, positionId)
-	if err != nil {
-		return sdk.Coins{}, sdk.Coins{}, err
+	if sender.String() != position.Address {
```

Comment:
Directly checking this instead of using `ensurePositionOwner` allows us to save unnecessary iteration over all positions in the pool